client/src/app/app.component.ts

```diff
@@ -40,35 +42,20 @@ export class AppComponent implements OnInit {
   }
 
   ngOnInit() {
-    this.activatedRoute.queryParams.subscribe(params => {
-      if (params['sso']) {
-        this.setSsoMode();
-      }
-    });
+    this.authenticationService.checkAndSetSsoMode();
 
     this.router.events.subscribe(event => {
       if (event instanceof NavigationStart) {
         if (event.url === '/') {
           this.loadAllProjects();
+        } else if (event instanceof NavigationStart && (event.url === '/login' || event.url === '/logout')) {
```

Comment:
``event instanceof NavigationStart && `` can be skipped here. It is checked earlier

Reply:
Done